app/components/primer/open_project/collapsible_helper.ts

```diff
@@ -0,0 +1,63 @@
+import {attr, controller, target, targets} from '@github/catalyst'
+
+@controller
+// eslint-disable-next-line custom-elements/expose-class-on-global
+export class CollapsibleHelperElement extends HTMLElement {
+  @target arrowDown: HTMLElement
+  @target arrowUp: HTMLElement
+  @targets collapsibleElements: HTMLElement[]
+
+  @attr collapsed: string
+  private _collapsed: boolean
+
+  connectedCallback() {
+    if (this.collapsed === 'true') {
+      this._collapsed = true
+      this.hideAll()
+    } else {
+      this._collapsed = false
+    }
+  }
+
+  toggle() {
+    if (this._collapsed) {
+      this._collapsed = false
+      this.expandAll()
+    } else {
+      this._collapsed = true
+      this.hideAll()
+    }
+  }
```

Comment:
If we are happy to make a `data-collapsed` a boolean attribute, this could be simplified as follows:

```suggestion
  @attr collapsed: boolean

  connectedCallback() {
    this.toggle()
  }

  toggle() {
    if (this.collapsed) {
      this.collapsed = false
      this.expandAll()
    } else {
      this.collapsed = true
      this.hideAll()
    }
  }
```

We could this more robust by defining `attributeChangedCallback`, so that the section is expanded/hidden even if other scripts modify the `collapsed` attribute.

```suggestion
  @attr collapsed = false

  toggle() {
    this.collapsed = !this.collapsed
  }

  attributeChangedCallback(name: string) {
    if (name === 'data-collapsed') {
      if (this.collapsed) {
        this.hideAll()
      } else {
        this.expandAll()
      }
    }
  }
```

https://github.com/user-attachments/assets/992868f7-bc07-41ee-ab9c-340f8776fe8c